src/test/java/org/folio/notes/controller/NoteTypesControllerTest.java

```diff
@@ -0,0 +1,347 @@
+package org.folio.notes.controller;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.emptyOrNullString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.ResultMatcher.matchAll;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.List;
+import java.util.UUID;
+import javax.persistence.EntityNotFoundException;
+import javax.validation.ConstraintViolationException;
+
+import org.hamcrest.Matcher;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.web.servlet.ResultMatcher;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+import org.folio.notes.domain.dto.NoteType;
+import org.folio.notes.domain.entity.NoteTypeEntity;
+import org.folio.notes.exception.NoteTypesLimitReached;
+import org.folio.notes.support.TestApiBase;
+import org.folio.spring.cql.CqlQueryValidationException;
+
+class NoteTypesControllerTest extends TestApiBase {
+
+  private static final String BASE_URL = "/note-types";
+
+  @Value("${folio.notes.types.default.limit}")
+  private String defaultNoteTypeLimit;
+
+  @BeforeEach
+  void setUp() {
+    setUpLimitConfigurationClient(defaultNoteTypeLimit);
+    databaseHelper.clearTable(TENANT);
+  }
+
+  // Tests for GET
+
+  @Test
+  @DisplayName("Find all note-types - empty collection")
+  void returnEmptyCollection() throws Exception {
+    mockMvc.perform(get("/note-types").headers(defaultHeaders()))
+      .andExpect(status().isOk())
+      .andExpect(jsonPath("$.noteTypes").doesNotHaveJsonPath())
+      .andExpect(jsonPath("$.totalRecords").value(0));
+  }
+
+  @Test
+  @DisplayName("Find all note-types")
+  void returnCollection() throws Exception {
+    List<NoteTypeEntity> noteTypes = List.of(
+      NoteTypeEntity.builder().name("first").build(),
+      NoteTypeEntity.builder().name("second").build()
+    );
+    databaseHelper.saveNoteTypes(noteTypes, TENANT);
```

Comment:
Extract creating of types collection to a method `List<NoteTypeEntity> createNoteTypes(List<String> typeNames)`.